crates/re_renderer/src/wgpu_resources/dynamic_resource_pool.rs

```diff
@@ -114,36 +157,39 @@ where
         // thread safety:
         // Since the count is pushed from 1 to 2 by `alloc`, it should not be possible to ever
         // get temporarily get back down to 1 without dropping the last user available copy of the Arc<Handle>.
-        self.alive_handles.retain(|handle, strong_handle| {
-            if Arc::strong_count(strong_handle) == 1 {
-                let desc = &self.resources[handle].0;
-                match self.last_frame_deallocated.entry(desc.clone()) {
-                    Entry::Occupied(mut e) => {
-                        e.get_mut().push(Arc::clone(strong_handle));
-                    }
-                    Entry::Vacant(e) => {
-                        e.insert(smallvec![Arc::clone(strong_handle)]);
+        state.alive_resources.retain(|_, resource| {
+            let resolved = resource
+                .take()
+                .expect("Alive resources should never be None");
```

Comment:
then why use an `Option` in `alive_resources`?? O.o

Reply:
ah ok, it needs to be taken here so we can unwrap it.

Reply:
yeah that whole thing is a bit of an annoying workaround